### ISSUE TITLE
[Tolino Shine 4] Backlight doesn't turn off completly

### ISSUE THREAD

**Issue by balu-** [koreader/koreader#11013]
* KOReader version:
  - 2023.08 "Lavender"
  - 2023.08-78 "Lavender"
* Device:
  - Tolino Shine 4 (should be the same hardware as kobo clara 2e) 

#### Issue

When switching of Background Lighting, the light dimes down but stays on - very suttle i had to go to a dark room to see it - but it's sill on. This is unfortunate because of battery usage.

#### Steps to reproduce

I bought a Tolino shine 4,
went to the debug menu (via searching for "1123581321")
and installed org.ds.simple.ink.launcher-1.2-release.apk and koreader-android-arm-v2023.08.apk . 

I opened a book and toucht the bottom left corner. 
Popup notification "Beleuchtung deaktiviert" is shown. 
But the background light stays on lightly.

Any hints what i can do or try to debug this behavior?
(with the stock tolino app background light can be switched off completly).

**Comment by pazos** [comment]
Your tolino must be using https://github.com/koreader/android-luajit-launcher/blob/master/app/src/main/java/org/koreader/launcher/device/lights/TolinoRootController.kt, which requires root to toggle the frontlight.

You can try to build yourself an APK with https://github.com/koreader/android-luajit-launcher/blob/master/app/src/main/java/org/koreader/launcher/device/lights/TolinoNtxController.kt and see if that works.

Anyhow, labeled as won't fix. We don't have any tolino. Patches welcome

**Comment by robintschoetschel** [comment]
I have a rooted Shine 4. When I select the "TolinoRootController" in the compatibility test, I can switch off the front light completely, but setting it using the slider doesn't turn it off completely. 

I think this might be related to an issue I opened in March: https://github.com/koreader/android-luajit-launcher/issues/410

I'd be happy to try and make this work, now having a bit more time for testing/coding again

**Comment by pazos** [comment]
@0x49D1: yup. Likebooks and Mebooks have no driver so they use the generic android way of dealing with lcd screens. That means no color temperature and no way to turn off the backlight completely.

**Comment by 0x49D1** [comment]
@pazos maybe someone with knowledge can guide me and I can just edit that generic driver to use specific SDK/parameters... I can't compile the build, but maybe can prepare the driver and help with build test. So: if someone, who knows how things work there, reads this and has no time to do the edits - willing to help here :)
